src/Microsoft.ComponentDetection.Common/ComponentStreamEnumerable.cs

```diff
@@ -1,29 +1,31 @@
 ﻿namespace Microsoft.ComponentDetection.Common;
+
 using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.IO;
 using Microsoft.ComponentDetection.Contracts;
+using Microsoft.Extensions.Logging;
 
 public class ComponentStreamEnumerable : IEnumerable<IComponentStream>
 {
+    private readonly ILogger logger;
```

Comment:
Why `ILogger` instead of `ILogger<ComponentStreamEnumerable>`?

Reply:
Because `ComponentStreamEnumerableFactory` gets a `ILogger<ComponentStreamEnumerableFactory>` injected into it, and then passes this logger directly into the constructor. `ComponentStreamEnumerable` isnt injected, its manually constructed.